src/extremeweatherbench/evaluate.py

```diff
@@ -174,8 +176,23 @@ def compute_case_operator(
         kwargs: Keyword arguments to pass to the metric computations.
 
     Returns:
-        A concatenated dataframe of the results of the case operator.
+        A concatenated dataframe of results from the case operator.
+
+    Raises:
+        TypeError: If any metric is not properly instantiated (i.e. isn't an instance
+        or child class of BaseMetric).
     """
+    # Validate that all metrics are instantiated (not classes or callables)
+    for i, metric in enumerate(case_operator.metric_list):
+        if isinstance(metric, type):
+            metric = metric()
+            case_operator.metric_list[i] = metric
```

Comment:
Would probably just do this in one line.

```suggestion
            case_operator.metric_list[i] = metric()
```